### ISSUE TITLE
Why are uint8 arrays casted to booleans?

### ISSUE THREAD

**Issue by banesullivan** [akaszynski/vtki#64]
https://github.com/akaszynski/vtki/blob/17826ca982e6382c370b3a8d4ae61cf44e46792d/vtki/common.py#L199-L200

@akaszynski, why is this in here?

**Comment by akaszynski** [comment]
You can’t load Boolean arrays directly into vtk objects, so you have to convert them to and from uint8.

**Comment by banesullivan** [comment]
Ahh, I see. Unfortunately, this comes at a price. I had some data arrays from using the `vtkTIFFReader` that produced a data array of `unit8` and this forced it to a boolean when I wrapped the output object:

```py
>>> import vtk
>>> import vtki
>>> from vtk.util.numpy_support import vtk_to_numpy
>>> reader = vtk.vtkTIFFReader()
>>> reader.SetFileName('sample.tif')
>>> reader.Update()
>>> image = vtki.wrap(reader.GetOutputDataObject(0))
>>> image.point_arrays['Tiff Scalars'].dtype
dtype('bool')
>>> arr = image.GetPointData().GetArray('Tiff Scalars')
>>> vtk_to_numpy(arr).dtype
dtype('uint8')


```

**Comment by akaszynski** [comment]
Ah, I can see why this is an issue.  I’ll have to make some changes internally to account for this issue and remove this from the code.  You can feel free to remove it if you wish.

**Comment by banesullivan** [comment]
What about [`vtkBitArray`](https://vtk.org/doc/nightly/html/classvtkBitArray.html)? We could send boolean arrays to this VTK type. I'm thinking to do this we'd need to implement our own `vtk_to_numpy` and `numpy_to_vtk` methods in the utilities to check the data types since the `numpy_support` module in VTK cannot handle boolean/bit data arrays. Here's an example of going back and forth between bit arrays and boolean arrays (I implemented something similar in PVGeo for string arrays which might be good to transfer here as well): 

```py
import vtk
from vtk.util.numpy_support import vtk_to_numpy, numpy_to_vtk
import numpy as np

# Create some arbitrary vtkBitArray
arr = vtk.vtkBitArray()
n = 10
arr.SetNumberOfTuples(n)
for i in range(n):
    arr.SetValue(i, i%2)

# Convert the vtkBitArray to NumPy boolean array
iarr = vtk.vtkCharArray()
iarr.DeepCopy(arr)
narr = np.frombuffer(iarr, dtype=bool)

# Convert NumPy boolean array back to a vtkBitArray
carr = numpy_to_vtk(narr.astype(np.uint8))
foo = vtk.vtkBitArray()
foo.DeepCopy(carr)

# Make sure the initial and final arrays are equal:
for i in range(10):
    assert foo.GetValue(i) == arr.GetValue(i)

```

**Comment by akaszynski** [comment]
This is probably the best approach since 
Fixed in dfacba2.
